lib/api/core/models/security/profile.js

```diff
@@ -21,23 +38,33 @@ function Profile () {
 Profile.prototype.isActionAllowed = function (requestObject, context, kuzzle) {
   var
     deferred = q.defer();
```

Comment:
Should be instantiated after the test returning `q(false)`. Or better yet: return the `this.getRoles` promise chain directly :-)